src/externalized/content/NPCQuoteInfo.cc

```diff
@@ -106,3 +106,136 @@ std::unique_ptr<NPCQuoteInfo const []> NPCQuoteInfo::deserialize(const JsonValue
+		obj.set("009factMustBeTrue", this->usFactMustBeTrue);
+	}
+	if (this->usFactMustBeFalse != FACT_NONE) {
+		obj.set("010factMustBeFalse", this->usFactMustBeFalse);
+	}
+
+	if (this->ubQuest != IRRELEVANT) {
+		JsonObject quest;
+		if (this->ubQuest > QUEST_DONE_NUM) {
+			quest.set("011index", this->ubQuest - QUEST_DONE_NUM);
+			quest.set("012status", "DONE");
+		}
+		else if (this->ubQuest > QUEST_NOT_STARTED_NUM) {
+			quest.set("013index", this->ubQuest - QUEST_NOT_STARTED_NUM);
+			quest.set("014status", "NOTSTARTED");
+		}
+		else {
+			quest.set("015index", this->ubQuest);
+			quest.set("016status", "INPROGRESS");
+		}
+		obj.set("017quest", quest.toValue());
+	}
+
+	if (this->ubFirstDay != 0) {
+		obj.set("018firstDay", this->ubFirstDay);
+	}
+	if (this->ubLastDay != IRRELEVANT) {
+		obj.set("019lastDay", this->ubLastDay);
+	}
+	if (this->ubApproachRequired != APPROACH_NONE) {
+		obj.set("020requiredApproach", Internals::getApproachName(static_cast<Approach>(this->ubApproachRequired)));
+	}
+	if (this->ubOpinionRequired != 0) {
+		obj.set("021requiredOpinion", this->ubOpinionRequired);
+	}
+
+	if (this->ubQuoteNum != IRRELEVANT) {
+		obj.set("022quoteNum", this->ubQuoteNum);
+	}
+	if (this->ubNumQuotes != IRRELEVANT) {
+		obj.set("023numQuotes", this->ubNumQuotes);
+	}
+
+	if (this->ubStartQuest != IRRELEVANT) {
+		obj.set("024startQuest", this->ubStartQuest);
+	}
+	if (this->ubEndQuest != IRRELEVANT) {
+		obj.set("025endQuest", this->ubEndQuest);
+	}
+
+	if (this->ubTriggerNPC != IRRELEVANT) {
+		// trigger closest merc who can see NPC
+		if (this->ubTriggerNPC == 0) obj.set("026triggerClosestMerc", true);
+		else if (this->ubTriggerNPC == 1) obj.set("027triggerSelf", true);
+		else obj.set("028triggerNPC", (contentManager->getMercProfileInfo(this->ubTriggerNPC))->internalName);
+	}
+	if (this->ubTriggerNPCRec != IRRELEVANT) {
+		obj.set("029triggerRecord", this->ubTriggerNPCRec);
+	}
+	if (this->usSetFactTrue != FACT_NONE) {
+		obj.set("030setFactTrue", this->usSetFactTrue);
+	}
+
+	if (this->usGiftItem != 0) {
+		const ItemModel* item = contentManager->getItem(this->usGiftItem);
+		if (item == nullptr) {
+			switch (this->usGiftItem)
+			{
+			case TURN_UI_OFF:
+				obj.set("031userInterface", "TURN_UI_OFF");
+				break;
+			case TURN_UI_ON:
+				obj.set("031userInterface", "TURN_UI_ON");
+				break;
+			case SPECIAL_TURN_UI_OFF:
+				obj.set("032userInterface", "SPECIAL_TURN_UI_OFF");
+				break;
+			case SPECIAL_TURN_UI_ON:
+				obj.set("033userInterface", "SPECIAL_TURN_UI_ON");
+				break;
+			default:
+				obj.set("034userInterface", ST::format("{}", this->usGiftItem));
+				break;
+			}
+		}
+		else {
+			obj.set("035giftItem", item->getInternalName());
+		}
+	}
+
+	if (this->usGoToGridno != NO_MOVE) {
+		obj.set("036goToGridno", this->usGoToGridno);
+	}
+
+	if (this->sActionData != NPC_ACTION_NONE) {
+		JsonObject actionData;
+		if (abs(this->sActionData) > 500 && abs(this->sActionData) < 650) {
```

Comment:
Would be better to use NPC_ACTION_TURN_TO_FACE_NEAREST_MERC and NPC_ACTION_LAST_TURN_TO_FACE_PROFILE